src/core/algorithms/gfd/gfd_miner.cpp

```diff
@@ -0,0 +1,683 @@
+#include "gfd_miner.h"
+
+#include <numeric>
+
+#include <boost/graph/copy.hpp>
+#include <boost/graph/isomorphism.hpp>
+#include <boost/graph/vf2_sub_graph_iso.hpp>
+#include <easylogging++.h>
+
+#include "config/option_using.h"
+
+namespace {
+
+using namespace algos;
+
+using Embedding = std::map<vertex_t, vertex_t>;
+using Embeddings = std::vector<Embedding>;
+
+using SingleRule = std::pair<std::vector<Literal>, Literal>;
+using Rule = std::pair<std::vector<Literal>, std::vector<Literal>>;
+using Rules = std::vector<Rule>;
+
+using Info = std::map<std::string, std::set<std::string>>;
+
+void NextSubset(std::vector<int>& indices, int const& border) {
+    if (indices.at(0) == border - static_cast<int>(indices.size())) {
+        indices = {};
+        return;
+    }
+    int index = 0;
+    for (int i = indices.size() - 1; i >= 0; --i) {
```

Comment:
If it is crucial to check from the last index be careful NOT to change it to size_t, due to overflow it might become an infinite cycle